### PR TITLE
DEX-1122 Fix locales for java.text.*Format

### PR DESCRIPTION
**Pull request by alexander-branevskiy**

### CODE REVIEW

File: dex/src/main/scala/com/wavesplatform/dex/api/http/entities/HttpOrderBook.scala [hunk 1]
```diff
@@ -43,7 +46,10 @@ object HttpOrderBook {
 
   }
 
-  private def formatValue(value: BigDecimal, decimals: Int): String = new java.text.DecimalFormat(s"0.${"0" * decimals}").format(value)
+  private def formatValue(value: BigDecimal, decimals: Int): String = {
+    val symbols = new DecimalFormatSymbols(Locale.US)
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/api/http/entities/HttpOrderBook.scala:50]: There is a lot of code inside the constructor. Let's cache the value somewhere.

File: dex/src/main/scala/com/wavesplatform/dex/settings/package.scala [hunk 2]
```diff
@@ -4,13 +4,16 @@ import com.typesafe.config.{Config, ConfigFactory}
...
+import java.text.{DecimalFormat, DecimalFormatSymbols}
+import java.util.Locale
 import scala.util.Try
 
 package object settings {
 
   implicit def toConfigOps(config: Config): ConfigOps = new ConfigOps(config)
 
-  private val format = new java.text.DecimalFormat("#.################")
+  private val symbols = new DecimalFormatSymbols(Locale.US)
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/settings/package.scala:15]: Too

File: dex/src/main/scala/com/wavesplatform/dex/settings/utils/rules.scala [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-import java.text.DecimalFormat
-
+import java.text.{DecimalFormat, DecimalFormatSymbols}
 import cats.implicits.{catsSyntaxOptionId, none}
 
+import java.util.Locale
+
 object rules {
-  private val doubleFormat = new DecimalFormat("#")
+  private val symbols = new DecimalFormatSymbols(Locale.US)
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/settings/utils/rules.scala:9]: Too

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex/src/main/scala/com/wavesplatform/dex/api/http/entities/HttpOrderBook.scala (hunk 1 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/api/http/entities/HttpOrderBook.scala]: resolved

File: dex/src/main/scala/com/wavesplatform/dex/settings/package.scala (hunk 2 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/settings/package.scala]: resolved

File: dex/src/main/scala/com/wavesplatform/dex/settings/utils/rules.scala (hunk 3 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/settings/utils/rules.scala]: resolved

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**